apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+
+const typeGenPatternKey = "*.ts";
+const typeGenPattern = "${capture}.typegen.ts";
+const hasTypeGenPattern = (fileNestingPatterns: object) =>
+  Object.values(fileNestingPatterns).some((pattern: string) => {
+    if (pattern.includes(",")) {
+      const patterns = pattern.split(",");
+      return patterns.some(
+        (innerPattern) => innerPattern.trim() === typeGenPattern
+      );
+    } else {
+      return pattern === typeGenPattern;
+    }
```

Comment:
nit: for simplicity of the check I would probably go with this:
```ts
    const patterns = pattern.split(",");
    return patterns.some(
      (innerPattern) => innerPattern.trim() === typeGenPattern
    );
```
It should behave in the same way

Reply:
I like it